From 1813c0d1ce5026d14a03a4f4c8a67f388e864940 Mon Sep 17 00:00:00 2001
From: Mark Yen <mark.yen@suse.com>
Date: Thu, 21 Jul 2022 11:07:40 -0700
Subject: [PATCH] Credentials: Linux: Allow overriding default credential
 store.

This adds a compile-time option to override the default credential store,
so that we can provide a different default for WSL installs of Rancher
Desktop.

Signed-off-by: Mark Yen <mark.yen@suse.com>
---
 Dockerfile                                    | 2 ++
 cli/config/credentials/default_store_linux.go | 8 ++++++++
 2 files changed, 10 insertions(+)

diff --git Dockerfile Dockerfile
index 8b0a1698ec..1fed30cc6a 100644
--- Dockerfile
+++ Dockerfile
@@ -50,6 +50,8 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     && /out/gotestsum --version

 FROM build-${BASE_VARIANT} AS build
+# GO_LDFLAGS defines additional linker arguments
+ARG GO_LDFLAGS
 # GO_LINKMODE defines if static or dynamic binary should be produced
 ARG GO_LINKMODE=static
 # GO_BUILDTAGS defines additional build tags
diff --git a/cli/config/credentials/default_store_linux.go b/cli/config/credentials/default_store_linux.go
index a9012c6d4a..1288694474 100644
--- a/cli/config/credentials/default_store_linux.go
+++ b/cli/config/credentials/default_store_linux.go
@@ -4,7 +4,15 @@ import (
 	"os/exec"
 )
 
+// LinuxCredentialsStoreDefaultOverride is a link-time override for the default
+// credentials store to use on Linux; this is set in the build scripts for
+// rancher-desktop-docker-cli for use in WSL.
+var LinuxCredentialsStoreDefaultOverride string
+
 func defaultCredentialsStore() string {
+	if LinuxCredentialsStoreDefaultOverride != "" {
+		return LinuxCredentialsStoreDefaultOverride
+	}
 	if _, err := exec.LookPath("pass"); err == nil {
 		return "pass"
 	}
-- 
2.32.0

